src/main/java/com/keshavarzi/airplanestats/security/controller/UserAuthorizationController.java

```diff
@@ -0,0 +1,73 @@
+@RequestMapping(path = "/api/user/authorization", name = "UserAuthorizationController")
+@AllArgsConstructor(onConstructor = @__(@Autowired))
+public class UserAuthorizationController {
+
+    private UserAuthorizationService userAuthorizationService;
+
+    /**
+     * Register a new user with email and password
+     * @param registerRequest DTO to create user
+     * @return HttpStatus.NOT_ACCEPTABLE: Invalid username or password
+     *
+     * <p>HttpStatus.CONFLICT: Existing Email in database (plane_stats.user_data.user)</p>
+     *
+     * <p>HttpStatus.NOT_FOUND: 'USER' Spring security role not present in database (plane_stats.user_data.role).
+     *      Roles should exist in database at all times</p>
+     *
+     * <p>HttpStatus.CREATED: User has been created with encoded password & stored in database (plane_stats.user_data.user)
+     *      as a 'USER' Spring security user</p>
+     */
+    @PostMapping(path = "register", name = "RegisterUser", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> register(@RequestBody RegisterRequest registerRequest) {
+        String email = registerRequest.getEmail();
+        try {
+            this.userAuthorizationService.register(email, registerRequest.getPassword());
+            return new ResponseEntity<>("User with email " + email + " has been created", HttpStatus.CREATED);
+        } catch (InvalidEmailException | InvalidPasswordException invalidCredentials) {
+            return new ResponseEntity<>(invalidCredentials.getMessage(), HttpStatus.NOT_ACCEPTABLE);
+        } catch (EmailExistException emailExistException) {
+            return new ResponseEntity<>(emailExistException.getMessage(), HttpStatus.CONFLICT);
+        } catch (AuthorizationRoleMissingException authorizationRoleMissingException) {
+            return new ResponseEntity<>(authorizationRoleMissingException.getMessage(), HttpStatus.NOT_FOUND);
+        }
+    }
+
+    @PostMapping(path = "login", name = "UserLogin",
+            consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<AuthorizationResponse> login(@RequestBody LoginRequest loginRequest) {
+        AuthorizationResponse authorizationResponse = new AuthorizationResponse();
+        try {
+            authorizationResponse =
+                    this.userAuthorizationService.login(loginRequest.getEmail(), loginRequest.getPassword());
+        } catch (UsernameNotFoundException usernameNotFoundException) {
+            authorizationResponse.setUnauthorizedError(usernameNotFoundException.getMessage());
+            return new ResponseEntity<>(authorizationResponse, HttpStatus.NOT_FOUND);
+        } catch (RuntimeException runtimeException) {
+            authorizationResponse.setUnauthorizedError(runtimeException.getMessage());
+            return new ResponseEntity<>(authorizationResponse, HttpStatus.UNAUTHORIZED);
+        }
+        return new ResponseEntity<>(authorizationResponse, HttpStatus.OK);
```

Comment:
why is there a return here, it can be on line 62 and removing the `authorizationResponse` variable

Reply:
Refactored